src/vcf/compression.cpp

```diff
@@ -106,9 +106,6 @@ namespace ebi
          * (5 characters), then we assume there's no compression.
          */
         if (line.size() < 5) {
```

Comment:
Are we completely sure this is the correct behavior? If the line is 4 characters it could still be compressed some way and have some (invalid) content.

In order to avoid the segmentation fault, we could check it inside the for loop as follows:

```cpp
for (auto & type : types) {
    /*
     * If the first line of the VCF file is shorter than the magic number string (stored in the 
     * variable "types"), the std::equal call below will cause a segmentation fault.
     */
    if (type.first.size() > line.length()) {
        continue;
    }

    if (std::equal(type.first.begin(), type.first.end(), line.begin())) {
        compressed_file_warning(type.second);
        return type.second;
    }

    return NO_EXT;
}
```

Reply:
hmm this sounds better.